src/components/catalog/CatalogCard.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import '../../css/catalog/CatalogCard.css';
+
+export default class CatalogCard extends Component {
+  constructor(props) {
+    super(props);
+
+    this.modalContainer = React.createRef();
+
+    this.state = {
+      course: {
+        title: '',
+        completedCourses: '',
+        totalCourses: '',
+        img: '',
+      },
+    };
+  }
+  static propTypes = {
+    course: PropTypes.object.isRequired,
```

Comment:
You should be more specific about the contents of `course`, so use `PropTypes.shape` instead of just `.object`
```suggestion
    course: PropTypes.shape({
      title: PropTypes.string,
      completed_courses: PropTypes.???,
      total_courses: PropTypes.???,
      img: PropTypes.???,
      lessons: PropTypes.arrayOf(PropTypes.object),
    }).isRequired,
```